marklogic-data-hub-central/ui-custom/src/components/RecentClear/RecentClear.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {useContext, useState} from "react";
+import "./RecentClear.scss";
+import ConfirmationModal from "../confirmation-modal/confirmation-modal";
+import { DetailContext } from "../../store/DetailContext";
+
+type Props = {
+    title?: string;
+};
+
+/**
+ * Component for showing a recent clear button around one or more other components.
+ *
+ * @component
+ * @prop {string} title RecentClear label.
+ * @example
+ * <RecentClear title="My RecentClear Title">
+ *   <SomeComponent />
+ * </RecentClear>
+ */
+
+const RecentClear: React.FC<Props> = (props) => {
+    const detailContext = useContext(DetailContext);
+    const [showModal, setShowModal] = useState(false);
+    const confirmationModalBody = "Clear recently visited record history for the current user?"
+
+    const showModalHandler = () => {
+      setShowModal(true);
+    }
+
+    const HandleDeleteRecords = async () => {
+      await detailContext.handleDeleteAllRecent();
+      setShowModal(false);
+    }
+
+    return (
+        <span className="recent-clear">
+         <button className="clear-button" data-testid="clearButton" onClick={showModalHandler}>{props.title}</button>
```

Comment:
As a UX improvement, can we make this button disabled if there are no recently viewed records to clear? I'd suggest writing a hasSavedRecords() method that checks for records for that user and use that method to disable if false. The Clear button should not be clickable if there are no records to clear.